libsolidity/formal/SMTEncoder.cpp

```diff
@@ -177,8 +177,12 @@ void SMTEncoder::visitFunctionOrModifier()
 	{
 		if (function.isImplemented())
 		{
+			auto const* currentFunction = m_variableUsage.getCurrentFunction();
+			solAssert(currentFunction, "");
+			m_variableUsage.setCurrentFunction(function);
```

Comment:
So BMC's function inlining ends up here?

Reply:
Yes, exactly.
The problem is that `VariableUsage` previously did not update what is the current function and then used incorrect `scopeContract` in `VariableUsage::endVisit(FunctionCall const&` leading to a lookup of a function in a contract that did not have it.